pkg/api/app_eth.go

```diff
@@ -0,0 +1,27 @@
+package api
+
+import (
+	"github.com/pkg/errors"
+	"github.com/wavesplatform/gowaves/pkg/proto"
+	"github.com/wavesplatform/gowaves/pkg/proto/ethabi"
+	"github.com/wavesplatform/gowaves/pkg/ride"
+	"github.com/wavesplatform/gowaves/pkg/state"
+)
+
+func (a *App) EthereumDAppMethods(addr proto.WavesAddress) (ethabi.MethodsMap, error) {
+	scriptInfo, err := a.state.ScriptInfoByAccount(proto.NewRecipientFromAddress(addr))
+	if err != nil {
+		if state.IsNotFound(err) {
```

Comment:
Why use this method? Is it no enough to use `errors.Is()`? This method `IsNotFound` doesn't use `errors.Is` inside, but checks only "upper" type of the error, doesn't look inside.

Reply:
State errors were refactored in last two commits.